eng/build.ps1

```diff
@@ -108,7 +118,30 @@ function Process-Arguments() {
        Print-Usage
        exit 0
     }
+    
+    if (!$vsBranch) {
+        if ($officialBuildId) {
+            Write-Host "vsBranch must be specified for official builds"
+            exit 1
+        }
+
+        $script:vsBranch = "dummy/ci"
+    }
 
+    if (!$vsDropName) {
+        if ($officialBuildId) {
+            Write-Host "vsDropName must be specified for official builds"
+            exit 1
+        }
+
+        $script:vsDropName = "Products/DummyDrop"
+    }
+
+    if (!$vsDropAccessToken -and $officialBuildId) {
+        Write-Host "vsDropAccessToken must be specified for official builds"
+        exit 1
+    }
```

Comment:
Could possibly simplify by having a single `if ($officialBulidId) {` and putting the other checks inside.

Reply:
It wouldn't be simpler since then I'd need to duplicate the checks for each of the variables.